client/state/preferences/actions.js

```diff
@@ -53,17 +70,29 @@ export const setPreference = ( key, value ) => ( {
  */
 export const savePreference = ( key, value ) => dispatch => {
 	dispatch( setPreference( key, value ) );
+	dispatch( {
+		type: PREFERENCES_SAVE,
+		key,
+		value
+	} );
+
 	const settings = {
 		[ USER_SETTING_KEY ]: {
 			[ key ]: value
 		}
 	};
 
-	return wpcom.me().settings().update( JSON.stringify( settings ) )
-		.then( ( data ) => {
-			dispatch( {
-				type: PREFERENCES_RECEIVE,
-				data
-			} );
+	return wpcom.me().settings().update( settings ).then( ( data ) => {
```

Comment:
The schema works too, but we fetch preferences after throwing out results 😄 
<img width="1241" alt="screen shot 2016-08-19 at 4 22 55 pm" src="https://cloud.githubusercontent.com/assets/1270189/17826818/621c788c-6629-11e6-9da0-b2e5d8b0e60c.png">